packages/compiler-cli/src/ngtsc/typecheck/extended/checks/missing_ngforof_let/index.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AST, TmplAstNode, TmplAstTemplate} from '@angular/compiler';
+import ts from 'typescript';
+
+import {ErrorCode, ExtendedTemplateDiagnosticName} from '../../../../diagnostics';
+import {NgTemplateDiagnostic} from '../../../api';
+import {TemplateCheckFactory, TemplateCheckWithVisitor, TemplateContext} from '../../api';
+
+/**
+ * Ensures a user doesn't forget to omit `let` when using ngfor.
+ * Will return diagnostic information when `let` is missing.
+ */
+class MissingNgForOfLetCheck extends TemplateCheckWithVisitor<ErrorCode.MISSING_NGFOROF_LET> {
+  override code = ErrorCode.MISSING_NGFOROF_LET as const;
+
+  override visitNode(
+      ctx: TemplateContext<ErrorCode.MISSING_NGFOROF_LET>, component: ts.ClassDeclaration,
+      node: TmplAstNode|AST): NgTemplateDiagnostic<ErrorCode.MISSING_NGFOROF_LET>[] {
+    const isTemplate = node instanceof TmplAstTemplate;
+    if (!(isTemplate) || (isTemplate && !(node.templateAttrs.length))) return [];
+    const attr = node.templateAttrs.find(x => x.name === 'ngForOf');
+    if (attr == null) return [];
+
+    if (node.variables.length > 0) return [];
```

Comment:
Style (in the compiler at least) is to always use blocks/braces with conditionals/loops/etc.

```suggestion
    if (node.variables.length > 0) {
      return [];
    }
```

Reply:
Fixed.